### PR TITLE
fix(@sap-ux/fe-mockserver-core): implicitly expand navigation properties used in a filter expression

### PR DESCRIPTION
**Pull request by drktfl**
This PR fixes a limitation on `$filter` expressions. In order to properly apply the filter condition, all data must be available at the time the filter is evaluated. This was only the case when the client requested that data by a corresponding `$expand` parameter.

That is,

`/Entities?$filter=navigationProperty2/value eq 0` and
`/Entities?$filter=navigationProperty2/value eq 0&$expand=navigationProperty2`
probably resulted in a different number of entries.

The mockserver now makes sure that the required data is available to the filters.

### CODE REVIEW

**Review by drktfl — COMMENTED** [review]
do not merge, needs some more work

File: packages/fe-mockserver-core/src/data/entitySets/entitySet.ts [hunk 1]
```diff
@@ -404,22 +401,19 @@ export class MockDataEntitySet implements EntitySetInterface {
...
-        if (lambdaOperator === 'ANY') {
-            return hasAnyValid;
-        } else {
-            return hasAllValid;
+            return this.checkFilter(mockDataToCheck, expression.expression, tenantId, odataRequest);
+        };
+
+        switch (expression.operator) {
+            case 'ALL':
+                return mockDataToCheckValue.every(check);
```
Comment (nlunets) [inline packages/fe-mockserver-core/src/data/entitySets/entitySet.ts:414]: this looks indeed better :P

File: packages/fe-mockserver-core/test/unit/v4/dataAccess.test.ts [hunk 2]
```diff
@@ -317,6 +317,106 @@ describe('Data Access', () => {
         expect(countryData.length).toEqual(1);
         expect(countryData[0].Name).toEqual('U S A');
     });
+    describe('additional $filter tests', () => {
```
Comment (nlunets) [inline packages/fe-mockserver-core/test/unit/v4/dataAccess.test.ts:320]: While this test is interesting to test the result of the filtering, i would also like to see a test for the `addExpandForFilters` case (maybe indirectly just through some test in the odataRequest) to validate that the filter expressions are properly taken into account, and also to check what happens when you have select + expand + filter.

**Review by nlunets — CHANGES_REQUESTED** [review]

**Review by nlunets — APPROVED** [review]

**Review by nlunets — APPROVED** [review]

File: packages/fe-mockserver-core/test/unit/v4/dataAccess.test.ts (hunk 2 above)
Comment (nlunets) [reply packages/fe-mockserver-core/test/unit/v4/dataAccess.test.ts:320]: Sounds good now

**Pull request merged**